src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2117,6 +2173,30 @@ internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType,
                     {
                         var stackAllocExpression = (BoundStackAllocArrayCreation)operand;
                         Error(diagnostics, ErrorCode.ERR_StackAllocConversionNotPossible, syntax, stackAllocExpression.ElementType, targetType);
+                        return;
+                    }
+                case BoundKind.UnconvertedSwitchExpression:
+                    {
+                        var switchExpression = (BoundUnconvertedSwitchExpression)operand;
```

Comment:
Nit: add variable to case declaration.

Reply:
The switch is on the bound kind, not the node.

---
In reply to: [301341912](https://github.com/dotnet/roslyn/pull/37052#discussion_r301341912) [](ancestors = 301341912)